server/src/main/java/org/candlepin/dto/api/v1/CertificateDTO.java

```diff
@@ -0,0 +1,167 @@
+
+import org.candlepin.dto.CandlepinDTO;
+
+import io.swagger.annotations.ApiModel;
+
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+
+
+/**
+ * The CertificateDTO is a DTO representing all Candlepin certificates presented to the API.
+ */
+@ApiModel(parent = TimestampedCandlepinDTO.class, description = "DTO representing a certificate")
+public class CertificateDTO extends TimestampedCandlepinDTO<CertificateDTO> {
+    public static final long serialVersionUID = 1L;
+
+    protected String id;
+    protected String key;
+    protected String cert;
+    protected CertificateSerialDTO serial;
+
+
+    /**
+     * Initializes a new CertificateDTO instance with null values.
+     */
+    public CertificateDTO() {
+        // Intentionally left empty
+    }
+
+    public String getId() {
+        return this.id;
+    }
+
+    public CertificateDTO setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    public String getKey() {
+        return this.key;
+    }
+
+    public CertificateDTO setKey(String key) {
+        this.key = key;
+        return this;
+    }
+
+    public String getCert() {
+        return this.cert;
+    }
+
+    public CertificateDTO setCert(String cert) {
+        this.cert = cert;
+        return this;
+    }
+
+    public CertificateSerialDTO getSerial() {
+        return this.serial;
+    }
+
+    public CertificateDTO setSerial(CertificateSerialDTO serial) {
+        this.serial = serial;
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        CertificateSerialDTO serial = this.getSerial();
+
+        return String.format("CertificateDTO [id: %s, key: %s, serial id: %s]",
+            this.getId(), this.getKey(), serial != null ? serial.getId() : null);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == this) {
+            return true;
+        }
+
+        if (obj instanceof CertificateDTO && super.equals(obj)) {
+            CertificateDTO that = (CertificateDTO) obj;
+
+            EqualsBuilder builder = new EqualsBuilder()
+                .append(this.getId(), that.getId())
+                .append(this.getKey(), that.getKey())
+                .append(this.getCert(), that.getCert())
+                .append(this.getSerial(), that.getSerial());
+
+            return builder.isEquals();
+        }
+
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int hashCode() {
+        HashCodeBuilder builder = new HashCodeBuilder(37, 7)
+            .append(super.hashCode())
+            .append(this.getId())
+            .append(this.getKey())
+            .append(this.getCert())
+            .append(this.getSerial());
+
+        return builder.toHashCode();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object clone() {
```

Comment:
Ahh, whoops. This got reverted when I was fighting with covarience earlier in the work -- I'll update that.

Regarding the copy constructor vs clone debate, I have nothing against copy constructors, but I generally prefer a clone method to them for a couple reasons:
- Not needing to know exactly which instance we need to instantiate to get a proper clone
- Not explicitly creating new instances in client code, which has potential to make mocking/testing a pain

What I'd like to do here is keep the clone method, fix the covarience and the obvious inefficiencies I overlooked (somehow forgot that base clone already makes a shallow copy for me), and add copy constructors which make use of the populate method. Would that be acceptable?